diff --git a/vowpalwabbit/Makefile.am b/vowpalwabbit/Makefile.am
index 1b5b75d6..1dcfce76 100644
--- a/vowpalwabbit/Makefile.am
+++ b/vowpalwabbit/Makefile.am
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = liballreduce.la libvw.la libvw_c_wrapper.la
+lib_LTLIBRARIES = liballreduce.la libvw.la
 
 liballreduce_la_SOURCES = allreduce_sockets.cc allreduce_threads.cc vw_exception.cc
 
@@ -17,7 +17,7 @@ ACLOCAL_AMFLAGS = -I acinclude.d
 AM_CXXFLAGS = ${BOOST_CPPFLAGS} ${ZLIB_CPPFLAGS} ${PTHREAD_CFLAGS} -Wall -Wno-unused-local-typedefs
 LIBS = ${BOOST_LDFLAGS} ${BOOST_PROGRAM_OPTIONS_LIB} ${ZLIB_LDFLAGS} ${PTHREAD_LIBS}
 
-CXXOPTIMIZE = 
+CXXOPTIMIZE =
 
 if PROFILE
 CXXOPTIMIZE += -pg
diff --git a/vowpalwabbit/gd.cc b/vowpalwabbit/gd.cc
index 7d747644..01ff2f83 100644
--- a/vowpalwabbit/gd.cc
+++ b/vowpalwabbit/gd.cc
@@ -587,6 +587,7 @@ template<class T>
 void save_load_regressor(vw& all, io_buf& model_file, bool read, bool text, T& weights)
 {
 	size_t brw = 1;
+  sync_weights(all);
 
 	if (all.print_invert)   //write readable model with feature names
 	{
@@ -732,7 +733,7 @@ void save_load_online_state(vw& all, io_buf& model_file, bool read, bool text, g
 void save_load_online_state(vw& all, io_buf& model_file, bool read, bool text, gd* g)
 { //vw& all = *g.all;
 	stringstream msg;
-
+  sync_weights(all);
 	msg << "initial_t " << all.initial_t << "\n";
 	bin_text_read_write_fixed(model_file, (char*)&all.initial_t, sizeof(all.initial_t),
 		"", read, msg, text);
