diff --git a/vowpalwabbit/gd.cc b/vowpalwabbit/gd.cc
index 464816a..0790c9d 100644
--- a/vowpalwabbit/gd.cc
+++ b/vowpalwabbit/gd.cc
@@ -692,6 +692,7 @@
     return;
   }

+  sync_weights(all);
   uint64_t i = 0;
   uint32_t old_i = 0;
   uint64_t length = (uint64_t)1 << all.num_bits;
@@ -825,6 +826,7 @@
   //vw& all = *g.all;
   stringstream msg;

+  sync_weights(all);
   msg << "initial_t " << all.initial_t << "\n";
   bin_text_read_write_fixed(model_file, (char*)&all.initial_t, sizeof(all.initial_t),
                             "", read, msg, text);
